migrations/src/postDeploymentSetup.js

```diff
@@ -44,23 +44,25 @@ module.exports = async () => {
   // Only transfer tokens in testnets
   if (NETWORK !== 'mainnet') {
     // Add new instance of StakeManager contract & Deployer address as Minter
-    await schellingCoin.addMinter(stakeManagerAddress);
+
     await schellingCoin.addMinter(signers[0].address);
+    const initialSupply = await schellingCoin.INITIAL_SUPPLY();
+    const stakeManagerSupply = (BigNumber.from(10).pow(BigNumber.from(26))).mul(BigNumber.from(6));
 
     if (SCHELLING_COIN_ADDRESS !== '') {
-      const { StakeManager: oldStakeManagerAddress } = await readOldDeploymentFile();
-
       // if previous instances of Schelling Coin is reused again and again,
       // then initial balance will get depleted, thus intial tokens minting is needed,
       // each time Schelling Coin instance is reused
-      const initialSupply = await schellingCoin.INITIAL_SUPPLY();
-      await schellingCoin.mint(signers[0].address, initialSupply);
+      await schellingCoin.addMinter(signers[0].address);
+      await schellingCoin.mint(signers[0].address, initialSupply - stakeManagerSupply);
 
-      // Remove previous instance of StakeManager contract & Deployer address from Minter
-      await schellingCoin.removeMinter(oldStakeManagerAddress);
+      // Remove previous instance of  Deployer address from Minter
       await schellingCoin.removeMinter(signers[0].address);
     }
 
+    // Remove previous instance of StakeManager contract & Deployer address from Minter
+    await schellingCoin.mint(stakeManagerAddress, initialSupply);
```

Comment:
Hey 
First we have to only mint mintableSupply to stakaManager, not initial supply
See
If shelling coin get deployed, deployer get complete Initial Supply minted.
Now again you mintinng here on top of it doesn't make sense.
It should be transfer

So  impl as per me would

```

 if (SCHELLING_COIN_ADDRESS !== '') { 
 Mint Intial Supply to Signers[0]
 } 
Transfer Mintable Supply from Signers[0] to StakeManager
```
This would work in both cases